drift/app.py

```diff
@@ -1,13 +1,51 @@
-import connexion
+import os
 import logging
 
+import connexion
+import watchtower
+from boto3.session import Session
+
 from drift import config
 from drift.views import v1
 from drift.error import handle_http_error
 from drift.exceptions import HTTPError
 from drift.metrics_registry import create_prometheus_registry_dir
 
 
+def _setup_cw_logging(logger):  # pragma: no cover
+    """
+    initialize cloudwatch logging
+
+    from https://github.com/RedHatInsights/cloudwatch-test
+    """
+    key_id = os.environ.get("CW_AWS_ACCESS_KEY_ID")
+    secret = os.environ.get("CW_AWS_SECRET_ACCESS_KEY")
+    if not (key_id and secret):
+        logger.info("CloudWatch logging disabled due to missing access key")
+        return
+
+    session = Session(
+        aws_access_key_id=key_id,
+        aws_secret_access_key=secret,
+        region_name=os.environ.get("AWS_REGION", "us-east-1"),
+    )
+
+    try:
+        with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r") as f:
```

Comment:
What is this reading?

Reply:
It is from https://docs.okd.io/latest/dev_guide/service_accounts.html#using-a-service-accounts-credentials-inside-a-container